app/models/imminence_response.rb

```diff
@@ -2,20 +2,42 @@ class ImminenceResponse
   INVALID_POSTCODE = "invalidPostcodeError".freeze
   NO_LOCATION = "validPostcodeNoLocation".freeze
 
-  attr_reader :places, :error, :postcode
+  attr_reader :places, :addresses, :error, :postcode
 
-  def initialize(postcode, places, error)
+  def initialize(postcode, imminence_response, error)
     @postcode = postcode
-    @places = places
     @error = error
+    @places = []
+    @addresses = []
+
+    return if @error
```

Comment:
Could we move this logic out of the initializer? Seems like it could live in the [place_controller](https://github.com/alphagov/frontend/pull/3525/files#diff-5d900c996c47b6c84232c18ba346309a47450e5bacdb2d2494b106c244cf9c58R88) and pass the variables `places and `addresses` in